services/repository/commit.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package repository
+
+import (
+	"context"
+	"fmt"
+
+	gitea_ctx "code.gitea.io/gitea/modules/context"
+	"code.gitea.io/gitea/modules/util"
+)
+
+type ContainedLinks struct { // TODO: better name?
+	Branches                 []*namedLink `json:"branches"`
+	Tags                     []*namedLink `json:"tags"`
+	ContainedInDefaultBranch bool         `json:"contained_in_default_branch"`
+	DefaultBranch            string       `json:"default_branch"`
+}
+
+type namedLink struct { // TODO: better name?
+	Name    string `json:"name"`
+	WebLink string `json:"web_url"`
```

Comment:
In code, "link" usually means "relative link" (eg: Repo.Link) while "url" means "full url" (eg: APIURL/HTMLURL)